lib/pbench/server/__init__.py

```diff
@@ -161,6 +165,24 @@ def _ref_datetime(self):
     def rest_uri(self):
         return self._get_conf("pbench-server", "rest_uri")
 
+    @property
+    def max_retention_period(self) -> timedelta:
+        """
+        Produce a timedelta representing the number of days the server allows
+        a dataset to be retained.
+
+        Returns
+            delta time representing retention period
+        """
+        try:
+            retention_days = int(
+                self._get_conf("pbench-server", "maximum-dataset-retention-days")
+            )
+        except (NoOptionError, NoSectionError):
+            retention_days = self.MAXIMUM_RETENTION_DAYS
```

Comment:
If we expect the "default" config file to have this value defined, should we fail if it is not there (since clearly, it's changed at this point by some other means)

Reply:
OK, I did look it up: [PBENCH-520](https://issues.redhat.com/browse/PBENCH-520) covers startup validation of config options. If we did that, we could fail on server startup for missing options rather than having fallback defaults ... if we really want to go that way.